hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1019,6 +1041,44 @@ func (r *HostedClusterReconciler) reconcileControlPlaneOperator(ctx context.Cont
 	return nil
 }
 
+func servicePublishingStrategyByType(hcp *hyperv1.HostedCluster, svcType hyperv1.ServiceType) *hyperv1.ServicePublishingStrategy {
```

Comment:
nit: it'd be nice to have a simple unit test for this.

Reply:
fixing....

Reply:
fixed